### ISSUE TITLE
Add support for drag images and drop descriptions

### ISSUE THREAD

**Issue by willibrandon** [gitextensions/gitextensions#11423]
### Feature description

Git Extensions supports drag and drop, but it displays the old school drag cursor.

https://github.com/gitextensions/gitextensions/assets/5017479/d96be9fe-6c02-4467-8e07-36d5408211ca

In .NET 7, Windows Forms added support for the new school drag cursor, allowing applications to display drag images and drop descriptions.

https://github.com/gitextensions/gitextensions/assets/5017479/0d7c45d6-1ea0-4486-9909-7f6ec07b62e4

If there is interest in adding support for drag images and drop descriptions to Git Extensions, I would be willing to draft a PR.

### Environment

- GitExtensions version: 33.33.33
- GIT version: 2.38.1.windows.1
- OS version: 10.0.22631
- .NET version: 8.0.0

**Comment by RussKie** [comment]
Oh, hey stranger 😀

**Comment by pmiossec** [comment]
> Oh, hey stranger 😀

Yes, I saw that maybe you could get some information on how it works from there : https://github.com/dotnet/winforms/pull/6576
🤣 🤣

**Comment by willibrandon** [comment]
My apologies for the delay. I've ran into a snag which I need to be upfront about before anyone wastes too much time on this.

Adding support for displaying the drag image when an item is dragged from Explorer is straightforward and works, however, the more interesting scenario is when `DoDragDrop` is called from the application with drag image support, which I've recently discovered, has some issues that I first need to resolve before I can consciously recommend implementing here.

https://github.com/dotnet/winforms/issues/8903
https://github.com/dotnet/winforms/pull/10319
https://github.com/dotnet/winforms/pull/10478

If Git Extensions is used by as many people as I think, I want to very careful about not introducing more issues for you all.

We can close this issue if you all agree that is appropriate, and re-open it when the fix gets released and is ready for prime-time.
